File: backend/app/models/case_contact.py
"""案件联系人模型"""
from sqlalchemy import Column, BigInteger, String, Boolean, DateTime, ForeignKey, JSON
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.core.database import Base


class CaseContact(Base):
    """案件联系人表"""
    __tablename__ = "case_contacts"

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    case_id = Column(BigInteger, ForeignKey('cases.id'), nullable=False, index=True, comment='案件ID')
    
    # 联系人信息
    contact_name = Column(String(100), nullable=False, comment='联系人姓名')
    phone_number = Column(String(50), nullable=False, comment='联系电话')
    relation = Column(String(50), nullable=False, comment='关系（本人/配偶/朋友/同事/家人等）')
    is_primary = Column(Boolean, default=False, index=True, comment='是否本人')
    
    # 可用渠道（JSON数组格式）
    available_channels = Column(JSON, comment='可用通信渠道，格式：["whatsapp", "sms", "call"]')
    
    # 备注
    remark = Column(String(500), comment='备注')
    
    # 时间戳
    created_at = Column(DateTime, server_default=func.now(), comment='创建时间')
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now(), comment='更新时间')

    # 关系
    case = relationship("Case", back_populates="contacts")
    communication_records = relationship("CommunicationRecord", back_populates="contact_person")

    class Config:
        orm_mode = True

File: backend-java/src/main/java/com/cco/model/entity/CaseContact.java
package com.cco.model.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 案件联系人实体类
 * 对应数据库表: case_contacts
 * 对应Python模型: app/models/case_contact.py
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("case_contacts")
public class CaseContact extends BaseEntity {
    
    @TableId(type = IdType.AUTO)
    private Long id;
    
    /**
     * 案件ID
     */
    private Long caseId;
    
    /**
     * 联系人姓名
     */
    private String contactName;
    
    /**
     * 联系电话
     */
    private String phoneNumber;
    
    /**
     * 关系（本人/配偶/朋友/同事/家人等）
     */
    private String relation;
    
    /**
     * 是否本人
     */
    private Boolean isPrimary;
    
    /**
     * 可用通信渠道（JSON格式）
     */
    private String availableChannels;
    
    /**
     * 备注
     */
    private String remark;
}












































